tracer/src/Datadog.Trace/Ci/CIEnvironmentValues.cs

```diff
@@ -912,6 +942,24 @@ private void SetupBuildkiteEnvironment()
             AuthorEmail = EnvironmentHelpers.GetEnvironmentVariable(Constants.BuildKiteBuildAuthorEmail);
             CommitterName = EnvironmentHelpers.GetEnvironmentVariable(Constants.BuildKiteBuildCreator);
             CommitterEmail = EnvironmentHelpers.GetEnvironmentVariable(Constants.BuildKiteBuildCreatorEmail);
+
+            // Node
+            NodeName = EnvironmentHelpers.GetEnvironmentVariable(Constants.BuildKiteAgentId);
+            var lstNodeLabels = new List<string>();
+            foreach (DictionaryEntry envvar in EnvironmentHelpers.GetEnvironmentVariables())
+            {
+                if (envvar.Key is string key && key.StartsWith(Constants.BuildKiteAgentMetadata, StringComparison.OrdinalIgnoreCase))
+                {
+                    var name = key.Replace(Constants.BuildKiteAgentMetadata, string.Empty)?.ToLowerInvariant();
```

Comment:
nit: is it worth doing a substring just in case of the _tiny_ possibility that the phrase appears later in the key too?
```suggestion
                    var name = key.Substring(Constants.BuildKiteAgentMetadata.Length).ToLowerInvariant();
```

Reply:
done in [df3a6a6](https://github.com/DataDog/dd-trace-dotnet/pull/4049/commits/df3a6a68b29d2d548bd006e1204e9c793b87c18e)